drivers/pwm/pwm_nrfx.c

```diff
@@ -176,6 +171,10 @@ static int pwm_nrfx_set_cycles(const struct device *dev, uint32_t channel,
 
 	seq_values_ptr_get(dev)[channel] = PWM_NRFX_CH_VALUE(compare_value, inverted);
 
+#ifdef CONFIG_NRFX_PWM120
+	sys_cache_data_flush_range((void *)config->seq.values.p_raw, config->seq.length);
```

Comment:
I think we need to check in runtime whether d-cache operation is needed or not, depending on `seq.values.p_raw` memory region attributes. You can take inspiration from this PR: https://github.com/zephyrproject-rtos/zephyr/pull/80015/files , see `UARTE_IS_CACHEABLE` .

Maybe in the future we could switch to `dmm` for this shim, but let's it keep simple as this is hotfix